### PR TITLE
deps: update djs to v14

### PR DESCRIPTION
**Pull request by skinmaker1345**
**Please describe the changes this PR makes and why it should be merged:**
This PR includes updating peer deps of djs to v14, thus making Dokdo compatible w/ discord.js v14.

**Status**
- [x] Code changes have been tested.

**Semantic versioning classification:**

- [ ] This PR includes new feature, methods or parameters
- [ ] This PR includes breaking changes (feature, methods, parameters removed or renamed)
- [ ] This PR **only** includes non-code(typo, documentation etc.) changes

**Notes**
testing.yml for Github Workflow was changed in favor of the changed dependency.

### CODE REVIEW

**Comment by wonderlandpark** [comment]
:wave: Thanks for contributing Dokdo!

File: src/commands/cat.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const fs = require('fs')
 const Discord = require('discord.js')
 const { ProcessManager, HLJS } = require('../utils')
+const { ButtonStyle } = require('discord.js')
```
Comment (wonderlandpark) [inline src/commands/cat.js:4]:
```suggestion
```

File: src/commands/cat.js [hunk 2]
```diff
@@ -11,9 +12,9 @@ module.exports = async function curl (message, parent) {
...
     await msg.init()
     await msg.addAction([
-      { button: new Discord.MessageButton().setStyle('DANGER').setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
-      { button: new Discord.MessageButton().setStyle('SECONDARY').setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
-      { button: new Discord.MessageButton().setStyle('SUCCESS').setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
+      { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
+      { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
+      { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Primary).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```
Comment (wonderlandpark) [inline src/commands/cat.js:17]:
```suggestion
      { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
      { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
      { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Primary).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```

File: src/commands/curl.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 const fetch = require('node-fetch')
 const Discord = require('discord.js')
 const { ProcessManager, HLJS } = require('../utils')
+const { ButtonStyle } = require('discord.js')
```
Comment (wonderlandpark) [inline src/commands/curl.js:4]:
```suggestion
```

File: src/commands/curl.js [hunk 4]
```diff
@@ -26,8 +27,8 @@ module.exports = async function curl (message, parent) {
   const msg = new ProcessManager(message, res || '', parent, { lang: type })
   await msg.init()
   await msg.addAction([
-    { button: new Discord.MessageButton().setStyle('DANGER').setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
-    { button: new Discord.MessageButton().setStyle('SECONDARY').setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
-    { button: new Discord.MessageButton().setStyle('SUCCESS').setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```
Comment (wonderlandpark) [inline src/commands/curl.js:32]:
```suggestion
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```

File: src/commands/exec.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
 const child = require('child_process')
 const Discord = require('discord.js')
 const { ProcessManager, codeBlock } = require('../utils')
+const { ButtonStyle } = require('discord.js')
```
Comment (wonderlandpark) [inline src/commands/exec.js:4]:
```suggestion
```

File: src/commands/exec.js [hunk 6]
```diff
@@ -20,9 +21,9 @@ module.exports = async function Exec (message, parent) {
   console.log(res.pid)
 
   await msg.addAction([
-    { button: new Discord.MessageButton().setStyle('DANGER').setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
```
Comment (wonderlandpark) [inline src/commands/exec.js:24]:
```suggestion
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
```

File: src/commands/exec.js [hunk 7]
```diff
@@ -20,9 +21,9 @@ module.exports = async function Exec (message, parent) {
...
     {
-      button: new Discord.MessageButton().setStyle('SECONDARY').setCustomId('dokdo$stop').setLabel('Stop'),
+      button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'),
```
Comment (wonderlandpark) [inline src/commands/exec.js:26]:
```suggestion
      button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'),
```

File: src/commands/exec.js [hunk 8]
```diff
@@ -31,7 +32,7 @@ module.exports = async function Exec (message, parent) {
         manager.destroy()
       }
     },
-    { button: new Discord.MessageButton().setStyle('SUCCESS').setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```
Comment (wonderlandpark) [inline src/commands/exec.js:35]:
```suggestion
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```

File: src/commands/js.js [hunk 9]
```diff
@@ -1,5 +1,6 @@
 const Discord = require('discord.js')
 const { ProcessManager, inspect, isinstance, isGenerator } = require('../utils')
+const { ButtonStyle } = require('discord.js')
```
Comment (wonderlandpark) [inline src/commands/js.js:3]:
```suggestion
```

File: src/commands/js.js [hunk 10]
```diff
@@ -52,8 +53,8 @@ module.exports = async function js (message, parent) {
...
   await msg.init()
   await msg.addAction([
-    { button: new Discord.MessageButton().setStyle('DANGER').setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
-    { button: new Discord.MessageButton().setStyle('SECONDARY').setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
-    { button: new Discord.MessageButton().setStyle('SUCCESS').setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```
Comment (wonderlandpark) [inline src/commands/js.js:58]:
```suggestion
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```

File: src/commands/jsi.js [hunk 11]
```diff
@@ -1,5 +1,6 @@
 const Discord = require('discord.js')
 const { ProcessManager, count, inspect, table, type } = require('../utils')
+const { ButtonStyle } = require('discord.js')
```
Comment (wonderlandpark) [inline src/commands/jsi.js:3]:
```suggestion
```

File: src/commands/jsi.js [hunk 12]
```diff
@@ -21,8 +22,8 @@ module.exports = async function jsi (message, parent) {
 
   await msg.init()
   await msg.addAction([
-    { button: new Discord.MessageButton().setStyle('DANGER').setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
-    { button: new Discord.MessageButton().setStyle('SECONDARY').setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
-    { button: new Discord.MessageButton().setStyle('SUCCESS').setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```
Comment (wonderlandpark) [inline src/commands/jsi.js:27]:
```suggestion
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Danger).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```

File: src/commands/shard.js [hunk 13]
```diff
@@ -1,5 +1,6 @@
 const Discord = require('discord.js')
 const { ProcessManager, inspect } = require('../utils')
+const { ButtonStyle } = require('discord.js')
```
Comment (wonderlandpark) [inline src/commands/shard.js:3]:
```suggestion
```

File: src/commands/shard.js [hunk 14]
```diff
@@ -23,8 +24,8 @@ module.exports = async function shard (message, parent) {
 
   await msg.init()
   await msg.addAction([
-    { button: new Discord.MessageButton().setStyle('DANGER').setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
-    { button: new Discord.MessageButton().setStyle('SECONDARY').setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
-    { button: new Discord.MessageButton().setStyle('SUCCESS').setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Primary).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
+    { button: new Discord.ButtonBuilder().setStyle(ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```
Comment (wonderlandpark) [inline src/commands/shard.js:29]:
```suggestion
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Primary).setCustomId('dokdo$prev').setLabel('Prev'), action: ({ manager }) => manager.previousPage(), requirePage: true },
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Secondary).setCustomId('dokdo$stop').setLabel('Stop'), action: ({ manager }) => manager.destroy(), requirePage: true },
    { button: new Discord.ButtonBuilder().setStyle(Discord.ButtonStyle.Success).setCustomId('dokdo$next').setLabel('Next'), action: ({ manager }) => manager.nextPage(), requirePage: true }
```

File: src/utils/ProcessManager.js [hunk 15]
```diff
@@ -172,6 +172,29 @@ module.exports = class ProcessManager {
...
+    const messages = []
+    let msg = ''
+    for (const chunk of splitText) {
+      if (msg && (msg + char + chunk).length > this.limit) {
+        messages.push(msg)
+        msg = ''
+      }
+      msg += (msg && msg !== '' ? char : '') + chunk
+    }
+    return messages.concat(msg).filter(m => m)
```
Comment (wonderlandpark) [inline src/utils/ProcessManager.js:198]: Is this code written by your self or is it from the original DJS source code?

File: src/utils/codeBlock.js [hunk 16]
```diff
@@ -7,7 +7,7 @@ module.exports = class codeBlock {
    */
   static construct (content, lang) {
     return `\`\`\`${content ? lang || '' : ''}
-${Util.escapeCodeBlock(content)}
+${escapeCodeBlock(content)}
```
Comment (wonderlandpark) [inline src/utils/codeBlock.js:10]: Btw looks like this code change isn't really related to this PR.

**Review by wonderlandpark — CHANGES_REQUESTED** [review]

**Comment by wonderlandpark** [comment]
+ looks like lock file needs a update (djsv14)

File: src/utils/ProcessManager.js (hunk 15 above)
Comment (skinmaker1345) [reply src/utils/ProcessManager.js:198]: It is from the original DJS code, removed unnecessary checks.

File: src/utils/codeBlock.js (hunk 16 above)
Comment (skinmaker1345) [reply src/utils/codeBlock.js:10]: Actually, discord.js removed Util. instead, the method which Util contained can be imported directly.

**Review by wonderlandpark — APPROVED** [review]
lgtm

**Pull request merged**